flask/app/email.py

```diff
@@ -0,0 +1,91 @@
+# using SendGrid's Python Library
+# https://github.com/sendgrid/sendgrid-python
+import os
+import json
+from datetime import datetime, timedelta
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import (
+    Asm,
+    Mail,
+    SendAt,
+    To,
+    From,
+    Subject,
+)
+from flask import current_app as app
+from pytz import timezone
+import math
+
+
+sg = SendGridAPIClient(os.environ.get("SENDGRID_API_KEY"))
```

Comment:
We use `getenv` elsewhere in the codebase, though they are totally aliases.

Reply:
> So I'm not quite seeing how email batching is implemented here, just a delay. I also don't see how the metadata dictionary becomes the email body. Are these both done on SendGrid?

Ah, the email batching (i.e. content from 5 analysis requests merged into 1 email) is not implemented yet. One email is sent per analysis request for now. I'll look into this!

Regarding the email body, this is done via SendGrid. What's happening is you have an HTML file that represents the email body and you set that as a template inside SendGrid. To tell your code to use that template, you can see that I have added a template ID.